go.mod

```diff
@@ -30,29 +30,29 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.11.0
 	github.com/prometheus/client_model v0.2.0
-	github.com/prometheus/common v0.26.0
+	github.com/prometheus/common v0.28.0
 	github.com/sirupsen/logrus v1.8.1
 	github.com/spf13/cobra v1.2.1
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
-	golang.org/x/net v0.0.0-20210825183410-e898025ed96a
+	golang.org/x/net v0.0.0-20211209124913-491a49abca63
 	golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
 	google.golang.org/grpc v1.40.0
 	gopkg.in/yaml.v2 v2.4.0
 	helm.sh/helm/v3 v3.6.1
-	k8s.io/api v0.22.2
-	k8s.io/apiextensions-apiserver v0.22.2
-	k8s.io/apimachinery v0.22.2
-	k8s.io/apiserver v0.22.2
-	k8s.io/client-go v0.22.2
-	k8s.io/code-generator v0.22.2
-	k8s.io/component-base v0.22.2
+	k8s.io/api v0.23.1
+	k8s.io/apiextensions-apiserver v0.23.0
+	k8s.io/apimachinery v0.23.1
+	k8s.io/apiserver v0.23.0
+	k8s.io/client-go v0.23.1
+	k8s.io/code-generator v0.23.1
+	k8s.io/component-base v0.23.1
 	k8s.io/klog v1.0.0
 	k8s.io/kube-aggregator v0.22.0
-	k8s.io/kube-openapi v0.0.0-20210527164424-3c818078ee3d
-	k8s.io/utils v0.0.0-20210819203725-bdf08cb9a70a
-	sigs.k8s.io/controller-runtime v0.10.0
-	sigs.k8s.io/controller-tools v0.6.2
+	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65
+	k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b
+	sigs.k8s.io/controller-runtime v0.11.0
```

Comment:
FYI - we [replace pin](https://github.com/operator-framework/operator-lifecycle-manager/blob/e2f2c5ba3b37898427a35e2e3fb9000d4781b4ee/go.mod#L248) the controller-runtime dependency to my fork due to how partial metadata objects are handled upstream right now.

Reply:
ok. Have you got an issue, something I could read to better understand the limitations?

Reply:
https://github.com/operator-framework/operator-lifecycle-manager/issues/2353 is the tracking issue but it's lacks any real context.